src/interface/viewport.c

```diff
@@ -844,6 +846,258 @@ void sub_0x69E8B0(uint32 eax, uint32 ecx){
+//because the wrong (because localized) string length is used for calculating the scroll position
+void viewport_banner_paint_setup(uint32_t ecx, int edx, rct_map_element* map_element)
+{
+	rct_drawpixelinfo* dpi = RCT2_GLOBAL(0x140E9A8, rct_drawpixelinfo*);
+	RCT2_GLOBAL(0x9DE570, uint8_t) = 0xC;
+	if (dpi->zoom_level > 1 || RCT2_GLOBAL(0x9DEA6F, uint8_t) & 1) return;
+	edx -= 16;
+	rct_scenery_entry* banner_scenery = g_bannerSceneryEntries[gBanners[map_element->properties.banner.index].type];
+	ecx += map_element->properties.banner.position;
+	ecx &= 3;
+	RCT2_GLOBAL(0x9DEA56, uint16_t) = edx + 2;
+	RCT2_GLOBAL(0x9DEA52, uint32_t) = RCT2_ADDRESS(0x98D884, uint32_t)[ecx * 2];
+	int ebx = (ecx << 1) + banner_scenery->image;
+	ebx += (gBanners[map_element->properties.banner.index].colour << 19) | 0x20000000;
+	if (map_element->flags & 0x10)
+	{
+		RCT2_GLOBAL(0x9DE570, uint8_t) = 0;
+		ebx &= 0x7FFFF;
+		ebx |= RCT2_ADDRESS(0x993CC4, uint32_t)[RCT2_GLOBAL(0x9AACBF, uint8_t)];
+	}
+	RCT2_CALLPROC_X(RCT2_ADDRESS(0x98197C, uint32_t)[RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_ROTATION, uint32_t)],
+		0x1500, ebx, 0, edx, 1, 1, RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_ROTATION, uint32_t));
+	RCT2_GLOBAL(0x9DEA52, uint32_t) = RCT2_ADDRESS(0x98D888, uint32_t)[ecx * 2];
+	ebx++;
+	RCT2_CALLPROC_X(RCT2_ADDRESS(0x98197C, uint32_t)[RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_ROTATION, uint32_t)],
+		0x1500, ebx, 0, edx, 1, 1, RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_ROTATION, uint32_t));
+	ecx ^= 2;
+	ecx--;
+	if (ecx >= 2 || (map_element->flags & 0x10)) return;
+	int ebp = banner_scenery->banner.var_06;
+	ebp += ecx;
+	RCT2_GLOBAL(RCT2_ADDRESS_COMMON_FORMAT_ARGS, uint32_t) = 0;
+	RCT2_GLOBAL(0x13CE956, uint32_t) = 0;
+	int eax = 0xBA5;
+	if (!(gBanners[map_element->properties.banner.index].flags & BANNER_FLAG_NO_ENTRY))
+	{
+		RCT2_GLOBAL(RCT2_ADDRESS_COMMON_FORMAT_ARGS, uint16_t) = gBanners[map_element->properties.banner.index].string_idx;
+		eax = 0x6C3;
+	}
+	format_string(RCT2_ADDRESS(RCT2_ADDRESS_COMMON_STRING_FORMAT_BUFFER, char), eax, (void*)RCT2_ADDRESS_COMMON_FORMAT_ARGS);
+	RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_FONT_SPRITE_BASE, uint16_t) = 0x1C0;
+	uint16_t string_width = gfx_get_string_width(RCT2_ADDRESS(RCT2_ADDRESS_COMMON_STRING_FORMAT_BUFFER, char));
+	uint16_t ticks = RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_TICKS, uint32_t) >> 1;
+	uint16_t cx = ticks % string_width;
+	int a = eax, b = ebx, c = cx, d = 0, si = 0, di = 0, p = ebp;
+	RCT2_CALLFUNC_X(0x6C42D9, &a, &b, &c, &d, &si, &di, &p);
+	RCT2_CALLPROC_X(RCT2_ADDRESS(0x98199C, uint32_t)[RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_ROTATION, uint32_t)],
+		0x1500, b, 0, edx + 22, 1, 1, RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_ROTATION, uint32_t));
+}
+
+//rct2: 0x0068B35F
+void sub_68B35F(int ax, int cx)
+{
+	if (ax < RCT2_GLOBAL(RCT2_ADDRESS_MAP_SIZE_UNITS, uint16_t) &&
+		cx < RCT2_GLOBAL(RCT2_ADDRESS_MAP_SIZE_UNITS, uint16_t) &&
+		ax >= 32 && cx >= 32)
+	{
+		RCT2_GLOBAL(0x141E9B4, uint32_t) = 0xFFFF;
```

Comment:
Yes. It's probably some kind of structure or something. I will have another look at it.